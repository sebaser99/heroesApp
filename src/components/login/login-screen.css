html, body, #root {
    height: 100%;
}
html {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

body {
    background-color: bisque;
    background-color: #FFE53B;
    background-image: linear-gradient(147deg, #FFE53B 0%, #FF2525 74%);

    
}
.contenedor {
    width: 100%;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
}
.box {
    background-color: #504a4a;
    max-width: 400px;
    padding: 40px 20px;
    text-align: center;
    color: white;
}
.box button{
    width: 100%;
}

