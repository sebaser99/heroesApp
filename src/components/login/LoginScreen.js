import { useContext } from "react"
import { useNavigate } from "react-router-dom"
import { AuthContext } from "../auth/authContext"
import { types } from "../types/types"
import './login-screen.css';

export const LoginScreen = () => {
    const navigate = useNavigate()
    const { dispatch} = useContext(AuthContext)
    

    const handleLogin = ()=>{
      
        const action = {
            type: types.login,
            payload:{
                name: "Sebastian Rodriguez"
            }
        }

        dispatch(action)

        let lastPath = localStorage.getItem('lastLocation') || '/marvel';
        
        navigate(lastPath, {
            replace: true
        });
    }

    return (
        <div className="contenedor">
            <div className="box">

                <h3 className="mb-4">Ingresa a Heroes App</h3>
                
                <button className='btn btn-warning'
                        onClick={handleLogin}>
                    Login
                </button>
                
            </div>

        </div>
    )
}
