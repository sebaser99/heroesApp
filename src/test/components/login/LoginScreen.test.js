import { mount } from "enzyme"
import { MemoryRouter, Route, Routes, useNavigate } from "react-router-dom"
import { AuthContext } from "../../../components/auth/authContext"
import { LoginScreen } from "../../../components/login/LoginScreen"
import { types } from "../../../components/types/types"

const mockNavigate = jest.fn()

    jest.mock('react-router-dom', ()=> ({
        ...jest.requireActual('react-router-dom'),
        useNavigate: ()=> mockNavigate
        }
    ))


describe('vamos a probar el <LoginScreen />', ()=>{
    

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false

        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter initialEntries={['/login']}>
                <Routes >
                    <Route path='/login'     element={<LoginScreen/>} />
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>)

        test('debe hacer match con el snapshot', ()=>{
            expect(wrapper).toMatchSnapshot()
        })
        test('debe realizar el dispatch y la navegaciÃ³n', ()=>{
             const handleClick = wrapper.find('button').prop('onClick');
             handleClick();
            expect(contextValue.dispatch).toHaveBeenCalledWith({
                type: types.login,
                payload:{
                    name: "Sebastian Rodriguez"
                }
            })

            expect(mockNavigate).toHaveBeenCalledWith('/marvel', {replace: true})

            localStorage.setItem('lastLocation', '/dc')
            handleClick()
          
            expect(mockNavigate).toHaveBeenCalledWith('/dc', {replace: true})

        })
})