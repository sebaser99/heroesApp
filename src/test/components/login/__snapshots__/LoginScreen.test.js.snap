// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vamos a probar el <LoginScreen /> debe hacer match con el snapshot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
